src/services/infra/DynamoDBService.ts

```diff
@@ -1,82 +1,50 @@
 
-const MOCK_LAUNCH: SiteLaunchMessage = {
-  repoName: "my-repo",
-  appId: "my-app",
-  primaryDomainSource: "example.com",
-  primaryDomainTarget: "myapp.example.com",
-  domainValidationSource: "example.com",
-  domainValidationTarget: "myapp.example.com",
-  requestorEmail: "john@example.com",
-  agencyEmail: "agency@example.com",
-  githubRedirectionUrl: "https://github.com/my-repo",
-  redirectionDomain: [
-    {
-      source: "example.com",
-      target: "myapp.example.com",
-      type: "A",
-    },
-  ],
-  status: { state: "pending", message: "PENDING_DURING_SITE_LAUNCH" },
-}
 export default class DynamoDBService {
   private readonly dynamoDBClient: DynamoDBClient
 
   private readonly TABLE_NAME: string
 
-  constructor() {
-    this.dynamoDBClient = new DynamoDBClient()
+  constructor(dynamoDBClient: DynamoDBClient) {
+    this.dynamoDBClient = dynamoDBClient
     this.TABLE_NAME = config.get("aws.dynamodb.siteLaunchTableName")
     autoBind(this)
   }
 
   async createItem(message: SiteLaunchMessage): Promise<void> {
-    await this.dynamoDBClient.createItem(this.TABLE_NAME, MOCK_LAUNCH)
+    await this.dynamoDBClient.createItem(this.TABLE_NAME, message)
   }
 
-  async getItem(message: SiteLaunchMessage): Promise<GetCommandOutput> {
-    return this.dynamoDBClient.getItem(this.TABLE_NAME, MOCK_LAUNCH.appId)
-  }
+  async getAllCompletedLaunches(): Promise<SiteLaunchMessage[]> {
+    const entries = ((
+      await this.dynamoDBClient.getAllItems(this.TABLE_NAME)
+    ).Items?.filter(isSiteLaunchMessage) as unknown) as SiteLaunchMessage[]
+
+    const completedEntries =
+      entries?.filter(
+        (entry) =>
+          entry.status?.state === "success" || entry.status?.state === "failure"
+      ) || []
 
-  async updateItem(message: SiteLaunchMessage): Promise<UpdateCommandOutput> {
-    // TODO: delete mocking after integration in IS-186
-    MOCK_LAUNCH.status = { state: "success", message: "SUCCESS_SITE_LIVE" }
-    const updateParams: UpdateParams = {
-      TableName: this.TABLE_NAME,
-      Key: { appId: MOCK_LAUNCH.appId },
-      // The update expression to apply to the item,
-      // in this case setting the "status" attribute to a value
-      UpdateExpression:
-        "set #status.#state = :state, #status.#message = :message",
-      ExpressionAttributeNames: {
-        "#status": "status",
-        "#state": "state",
-        "#message": "message",
-      },
-      // A map of expression attribute values used in the update expression,
-      // in this case mapping ":state" to the value of the Launch status state and ":message" to the value of the Launch status message
-      ExpressionAttributeValues: {
-        ":state": MOCK_LAUNCH.status.state,
-        ":message": MOCK_LAUNCH.status.message,
-      },
-    }
-    return this.dynamoDBClient.updateItem(updateParams)
+    // Delete after retrieving the items
+    Promise.all(completedEntries.map((entry) => this.deleteItem(entry)))
```

Comment:
can't see the previous comments here but my concern was that this flow: 

1. retrieve entries
2. delete entries
3. send email to ops 

over 

1. retrieve entries
2. send email to ops
3. delete entries where email sent successfully

means that we dk failures till the user writes in whereas if we delete after email sending, we can always check the db to see if there are any failures and take action early

Reply:
hmm, you might be misunderstanding me here - my concern isn't that the emails aren't sent, but that **failed launches can be silently ignored**. 

ie, if the email sending fails, no one will know until the agency writes in about their failed launch whereas if we don't delete until email sending is successful, we can check the db (at the end) to see if there are any failed launches. because most failed launches are human errors, my thinking was that just checking the db at the end then telling the user to restart the process might be a better flow than waiting for them to write in

Reply:
> ie, if the email sending fails, no one will know until the agency writes in about their failed launch whereas if we don't delete until email sending is successful, we can check the db (at the end) to see if there are any failed launches. because most failed launches are human errors, my thinking was that just checking the db at the end then telling the user to restart the process might be a better flow than waiting for them to write in

Hey yup, you are right to say that this flow may allow launches to fail silently! However, in this case, I would opt for keeping our DB table lean as: 

1. Operationally, agencies + ops tend to be quite anxious about site launch, and typically contact us within 4 hours if the site is not up. It is unlikely that a "www" site is due for launch and eng is not aware of it by the end of the day. 
2. I do think it is going to be unnecessary overhead to ask an on-call to periodically check dynamo DB on the days of site launch. 
3. To check for the cause of failure, eng can look at the SOT for launches, ie amplify console and ask the user to restart the process. 
4. There needs to be a way to mark a successful consumption of the row due to the existence of multiple BE instances. 

I think can chat offline uh text is hard HAHA

Reply:
marking this as resolved - even though we have silent failures, the db is just acting as a message broker and it doesn't handle anything beyond message storage + delivery. 

we'll check elsewhere (amp console) for launch failures and recover there instead